File: installer/src/test/java/com/teamscale/profiler/installer/MockRegistry.java
package com.teamscale.profiler.installer;

import com.teamscale.profiler.installer.windows.IRegistry;
import com.teamscale.profiler.installer.windows.WindowsRegistry;

import java.util.HashMap;
import java.util.Map;

/**
 * Mock of {@link IRegistry} to allow tests without actually changing the Windows registry.
 */
public class MockRegistry implements IRegistry {

	private final Map<String, String> values = new HashMap<>();

	@Override
	public String getHklmValue(String key, String name) {
		return values.get(key + "\\" + name);
	}

	@Override
	public void setHklmValue(String key, String name, String value) {
		values.put(key + "\\" + name, value);
	}

	@Override
	public void deleteHklmValue(String key, String name) {
		values.remove(key + "\\" + name);
	}

	/** Reads the given environment variable from the registry. */
	public String getVariable(String name) {
		return getHklmValue(WindowsRegistry.ENVIRONMENT_REGISTRY_KEY, name);
	}

	/** Sets the given environment variable in the registry. */
	public void setVariable(String name, String value) {
		setHklmValue(WindowsRegistry.ENVIRONMENT_REGISTRY_KEY, name, value);
	}
}

File: installer/src/test/kotlin/com/teamscale/profiler/installer/utils/MockRegistry.kt
package com.teamscale.profiler.installer.utils

import com.teamscale.profiler.installer.windows.IRegistry
import com.teamscale.profiler.installer.windows.WindowsRegistry

/**
 * Mock of [IRegistry] to allow tests without actually changing the Windows registry.
 */
class MockRegistry : IRegistry {
	private val values = mutableMapOf<String, String>()

	override fun getHklmValue(key: String, name: String) = values["$key\\$name"]

	override fun setHklmValue(key: String, name: String, value: String) {
		values["$key\\$name"] = value
	}

	override fun deleteHklmValue(key: String, name: String) {
		values.remove("$key\\$name")
	}

	/** Reads the given environment variable from the registry.  */
	fun getVariable(name: String) =
		getHklmValue(WindowsRegistry.ENVIRONMENT_REGISTRY_KEY, name)

	/** Sets the given environment variable in the registry.  */
	fun setVariable(name: String, value: String) {
		setHklmValue(WindowsRegistry.ENVIRONMENT_REGISTRY_KEY, name, value)
	}
}
